File: app/src/main/java/cf/movie/slmovie/main/newMovies/model/NewMoviesPagerAdapter.java
package cf.movie.slmovie.main.newMovies.model;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import cf.movie.slmovie.base.BaseMovies.constant.Which;
import cf.movie.slmovie.base.BaseMovies.ui.BaseMoviesFragment;

/**
 * Created by 包俊 on 2017/7/22.
 */

public class NewMoviesPagerAdapter extends FragmentPagerAdapter {
    private static final int PAGE_COUNT = 8;

    public NewMoviesPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.LastUpdateMovie);
            case 1:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.ActionMovie);
            case 2:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.Comedy);
            case 3:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.LoveMovie);
            case 4:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.ScienceMovie);
            case 5:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.HorrorMovie);
            case 6:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.DramaMovie);
            case 7:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.WarMovie);
            default:
                return BaseMoviesFragment.Companion.newInstance(Which.UrlType.LastUpdateMovie);
        }
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return Which.INSTANCE.getLastUpdateMovieStr();
            case 1:
                return Which.INSTANCE.getActionMovieStr();
            case 2:
                return Which.INSTANCE.getComedyStr();
            case 3:
                return Which.INSTANCE.getLoveMovieStr();
            case 4:
                return Which.INSTANCE.getScienceMovieStr();
            case 5:
                return Which.INSTANCE.getHorrorMovieStr();
            case 6:
                return Which.INSTANCE.getDramaMovieStr();
            case 7:
                return Which.INSTANCE.getWarMovieStr();
            default:
                return Which.INSTANCE.getLastUpdateMovieStr();
        }
    }

}

File: app/src/main/java/cf/movie/slmovie/main/newMovies/model/NewMoviesPagerAdapter.kt
package cf.movie.slmovie.main.newMovies.model

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import cf.movie.slmovie.base.BaseMovies.constant.Which
import cf.movie.slmovie.base.BaseMovies.ui.BaseMoviesFragment

/**
 * Created by 包俊 on 2017/7/22.
 */

class NewMoviesPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return BaseMoviesFragment.newInstance(Which.UrlType.LastUpdateMovie)
            1 -> return BaseMoviesFragment.newInstance(Which.UrlType.ActionMovie)
            2 -> return BaseMoviesFragment.newInstance(Which.UrlType.Comedy)
            3 -> return BaseMoviesFragment.newInstance(Which.UrlType.LoveMovie)
            4 -> return BaseMoviesFragment.newInstance(Which.UrlType.ScienceMovie)
            5 -> return BaseMoviesFragment.newInstance(Which.UrlType.HorrorMovie)
            6 -> return BaseMoviesFragment.newInstance(Which.UrlType.DramaMovie)
            7 -> return BaseMoviesFragment.newInstance(Which.UrlType.WarMovie)
            else -> return BaseMoviesFragment.newInstance(Which.UrlType.LastUpdateMovie)
        }
    }

    override fun getCount(): Int {
        return PAGE_COUNT
    }

    override fun getPageTitle(position: Int): CharSequence {
        when (position) {
            0 -> return Which.LastUpdateMovieStr
            1 -> return Which.ActionMovieStr
            2 -> return Which.ComedyStr
            3 -> return Which.LoveMovieStr
            4 -> return Which.ScienceMovieStr
            5 -> return Which.HorrorMovieStr
            6 -> return Which.DramaMovieStr
            7 -> return Which.WarMovieStr
            else -> return Which.LastUpdateMovieStr
        }
    }

    companion object {
        private val PAGE_COUNT = 8
    }

}
